waspc/packages/wasp-config/__tests__/testFixtures.ts

```diff
@@ -1,48 +1,122 @@
 /**
  * This module contains sample data that can be used for testing purposes.
- * In our case the sample data represents UserSpec data.
+ * In our case the sample data represents TsAppSpec data.
  */
 
 import * as AppSpec from "../src/appSpec.js";
 import { Branded } from "../src/branded.js";
-import * as UserApi from "../src/userApi.js";
+import { App } from "../src/publicApi/App.js";
+import * as TsAppSpec from "../src/publicApi/tsAppSpec.js";
 
 /**
- * Creates a type containing only the required properties from T.
+ * Creates a type containing only the required properties from T recursively.
  *
  * This utility:
- * - Filters out optional properties from the type
- * - Provides a clean type for minimal configuration objects
- * - Returns `Record<string, never>` (empty object) when no required properties exist
+ * - Filters out optional properties recurisvely
+ * - Stops from unwrapping `Branded` types
+ * - Returns `EmptyObject` when no required properties exist
  *
  * @template T - The type to extract required properties from
+ *
+ * @example
+ * ```ts
+ * // Given the following type:
+ * type Result = MinimalConfig<{
+ *   a: Branded<string, "A">;
+ *   b: {
+ *     c: boolean;
+ *     d?: {
+ *       e: boolean;
+ *     };
+ *   };
+ *   f: {
+ *     g: boolean;
+ *     h?: string;
+ *   }[];
+ * };
+ * // The result will be:
+ * type Result = {
+ *   a: Branded<string, "A">;
+ *   b: {
+ *     c: boolean;
+ *   };
+ *   f: {
+ *     g: boolean;
+ *     h: string;
+ *   }[];
+ * };
+ */
+type MinimalConfig<T> =
+  T extends Branded<infer TType, infer TBrand>
+    ? Branded<TType, TBrand>
+    : T extends Array<infer TArrayItem>
+      ? Array<MinimalConfig<TArrayItem>>
+      : T extends object
+        ? keyof T extends never
+          ? EmptyObject
+          : {
+              [K in keyof T as T[K] extends Required<T>[K]
+                ? K
+                : never]: MinimalConfig<T[K]>;
+            }
+        : T;
+
+/**
+ * Represents an empty object type in TypeScript.
  * @see https://www.totaltypescript.com/the-empty-object-type-in-typescript
  */
-type MinimalConfig<T> = keyof T extends never
-  ? Record<string, never>
-  : {
-        [K in keyof T as undefined extends T[K] ? never : K]: T[K];
-      } extends infer R
-    ? keyof R extends never
-      ? Record<string, never>
-      : R
-    : never;
+type EmptyObject = Record<string, never>;
 
 /**
- * Creates a type with all properties and nested properties required.
+ * Creates a type with all properties of T required recursively.
  *
  * This utility:
- * - Makes all properties required recursively (removes optional flags)
- * - Stops from unwrapping branded types fully
+ * - Makes all properties required recursively
+ * - Stops from unwrapping branded types
  *
  * @template T - The type to make fully required
+ *
+ * @example
+ * ```ts
+ * // Given the following type:
```

Comment:
The examples are great.

I mentioned in a different comment that I'm leaning towards putting them in real tests.